src/OpenTelemetry.Contrib.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -42,9 +46,33 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            activity.DisplayName = this.GetDisplayName(activity);
+            string displayName = this.GetDisplayName(activity);
+            ActivityContext activityContext = default;
+            Activity originalActivityCurrent = Activity.Current;
+            if (activity.ParentSpanId == default)
+            {
+                Activity.Current = null;
+            }
+            else
+            {
+                activityContext = new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true);
+            }
+
+            activity = ActivitySource.StartActivity(activity.OperationName, this.GetActivityKind(originalActivityCurrent), activityContext);
+
+            if (activity == null)
+            {
+                Activity.Current = originalActivityCurrent;
+            }
+            else
+            {
+                foreach (var tag in originalActivityCurrent.TagObjects)
```

Comment:
`originalActivityCurrent` could be null here

Reply:
fixed